src/RTTComms.cpp

```diff
@@ -431,8 +431,13 @@ namespace BrainCloud
                                 host += key + "=" + value;
                             }
                         }
-
+                        // ############################
+                        // temporary fix for Android
+                        // this will cause a seg fault (nullptr access) in Android if RTT is enabled
+                    #if defined(__ANDROID__) and defined(USE_LIBWEBSOCKETS)
```

Comment:
Lets put a TODO, here, and further investigate android and further tests. I'm averse to accepting these changes due to this.

Reply:
If the following line is there, then there is an undefined reference.

 _socket = IWebSocket::create(host, port, headers);

The required file is included in cmaklists only if use_libwebsockets is defined:
# Sockets
if (USE_LIBWEBSOCKETS)
    list(APPEND OS_SPECIFIC_INCS
            "include/braincloud/internal/DefaultWebSocket.h")
    list(APPEND OS_SPECIFIC_SRCS "src/DefaultWebSocket.cpp")
endif() 

libwebsockets cannot build with curl/openssl because it is using mbedtls. 

A solution may be to rebuild curl with mbedtls instead of openssl. A solution may be to figure out how to use both in the same project (without multiple definitions). A solution may also be to use the other type of ITCPSocket as endpoint.

Reply:
When RTT is enabled, the user will get "No endpoint available". There is no null pointer crash, just RTT does not work.